protocols/kad/src/behaviour.rs

```diff
@@ -452,11 +563,17 @@ impl<TSubstream> Kademlia<TSubstream> {
             _ => {}
         }
     }
+
+    #[cfg(test)]
+    fn has_record(&self, key: &Multihash) -> bool {
+        self.records.get(key).is_some()
+    }
 }
 
-impl<TSubstream> NetworkBehaviour for Kademlia<TSubstream>
+impl<TSubstream, TRecordStorage> NetworkBehaviour for Kademlia<TSubstream, TRecordStorage>
 where
     TSubstream: AsyncRead + AsyncWrite,
+    TRecordStorage: RecordStore + Default,
```

Comment:
Why the `Default` constraint here?

Reply:
The trait bound from the [this impl block for Kademlia](https://github.com/montekki/rust-libp2p/blob/fs-kademlia-session-key-lookups/protocols/kad/src/behaviour.rs#L215) brings it here since we use methods from it.

Reply:
It should be removed from both impl blocks, and added back only where relevant.
```suggestion
    TRecordStorage: RecordStore,
```